lib/facebook_ads/model.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+require 'facebook_ads/json_converter'
+
+module FacebookAds
+  class Model
+    class << self
+      def field_converters
+        @field_converters ||= {}
+      end
+
+      def known_fields
+        field_converters.keys
+      end
+
+      def field(name, type: nil)
+        name = name.to_s
+        field_converters[name] = JsonConverter.for(type)
+
+        define_method name do
+          fields[name]
+        end
+
+        define_method "#{name}=" do |value|
+          fields[name] = value
+        end
+      end
+    end
+
+    attr_reader :id, :fields, :client
+
+    def initialize(id, client: FacebookAds.client)
+      @id = id
+      @fields = {}
+      @client = client
+    end
+
+    def fetch(fields)
+      fields = Array(fields)
+      validate_fields(fields)
+
+      response = client.get(id, params: { fields: fields.join(',') })
+      response.each do |field, value|
+        converter = self.class.field_converters[field]
+        next if converter.nil?
```

Comment:
I'd prefer to keep it simple now and start to hate the API in the future, when we find out that it behaves in this annoying way... (and then do the check of course)